packages/inject/src/inject.ts

```diff
@@ -0,0 +1,213 @@
+import { Config, WebchatEvent, WebchatEventType } from '@botpress/webchat'
+
+require('./inject.css')
+
+interface WebchatRef {
+  postMessage: (message: any, targetOrigin: string) => void
+  eventListener: {
+    handler: (event: WebchatEvent) => void
+    topics: WebchatEventHandlerTopics
+  }
+}
+
+type WebchatEventHandler = (evnt: WebchatEvent) => void
+type WebchatEventHandlerTopics = WebchatEventType[] | ['*']
+
+// full backward compatibility
+const DEFAULT_CHAT_ID = 'bp-web-widget'
+const DEFAULT_IFRAME_ID = 'bp-widget'
+
+const CHAT_REFS: { [chatId: string]: WebchatRef } = {}
+
+function _getContainerId(chatId?: string) {
+  return chatId ? `${chatId}-container` : DEFAULT_CHAT_ID
+}
+
+function _getIframeId(chatId: string) {
+  return chatId || DEFAULT_IFRAME_ID
+}
+
+function _injectDOMElement(
+  tagName: string,
+  selector: string,
+  options: { [key: string]: string } = {}
+): HTMLElement | void {
+  const element = document.createElement(tagName)
+  // @ts-ignore
+  Object.entries(options).forEach(([attrName, attrValue]) => (element[attrName] = attrValue))
+  const parent = document.querySelector(selector)
+  if (!parent) {
+    console.error(`No element correspond to ${selector}`)
```

Comment:
Shouldn't we throw here as this might be a "critical" error depending on the context?

Reply:
done